### PR TITLE
Add metabomics-lcms-preprocessing WF

### PR DESCRIPTION
**Pull request by lecorguille**
Here is a first WF for metabolomics link to this [GTN](https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html).

It would be great to have it in WorkflowHub.eu. We plan a paper which will refere to.

### CODE REVIEW

**Comment by mvdbeek** [comment]
Thanks @lecorguille, can you please have a look at the steps in https://github.com/galaxyproject/iwc/blob/main/workflows/README.md#adding-workflows ?

**Comment by lecorguille** [comment]
> Thanks @lecorguille, can you please have a look at the steps in https://github.com/galaxyproject/iwc/blob/main/workflows/README.md#adding-workflows ?

Should be ok now?

**Comment by mvdbeek** [comment]
Yes, looks better but you're still missing the readme and changelog.

**Comment by lldelisle** [comment]
Also, could you rename the `_test.yml` to `-tests.yml`? Thanks

File: workflows/metabomics-lcms-preprocessing/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Mass spectrometry: LC-MS preprocessing with XCMS 
+
+This workflow is composed with the XCMS tool R package [(Smith, C.A. 2006)](https://bioconductor.org/packages/release/bioc/html/xcms.html) able to extract, filter, align and fill gapand the possibility to annotate isotopes, adducts and fragments using the CAMERA R package [(Kuhl, C 2012)](https://bioconductor.org/packages/release/bioc/html/CAMERA.html).
```
Comment (mvdbeek) [inline workflows/metabomics-lcms-preprocessing/README.md:3]:
```suggestion
This workflow uses the XCMS tool R package [(Smith, C.A. 2006)](https://bioconductor.org/packages/release/bioc/html/xcms.html) to extract, filter, align and fill gaps, and uses the CAMERA R package [(Kuhl, C 2012)](https://bioconductor.org/packages/release/bioc/html/CAMERA.html) to annotate isotopes, adducts and fragments .
```

File: workflows/metabomics-lcms-preprocessing/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+🎓 For more information, it is explained in the [Galaxy Training Network tutorial: Mass spectrometry: LC-MS preprocessing with XCMS](https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html)
```
Comment (mvdbeek) [inline workflows/metabomics-lcms-preprocessing/README.md:5]:
```suggestion
🎓 For more information see the [Galaxy Training Network tutorial: Mass spectrometry: LC-MS preprocessing with XCMS](https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html)
```

File: workflows/metabomics-lcms-preprocessing/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+A sample metadata file contains various information for each of your raw files:
+- Classes which will be used during the preprocessing steps
+- Analytical batches which will be useful for a batch correction step, along with sample types (pool/sample) and injection order
+- Different experimental conditions which can be used for statistics
+- Any information about samples that you want to keep, in a column format
+
+The content of your sample metadata file has to be filled by you, since it is not contained in your raw data. Note that you can either:
+- Upload an existing metadata file
+- Use a template to create one (because it can be painful to get the sample list without misspelling or omission)
```
Comment (mvdbeek) [inline workflows/metabomics-lcms-preprocessing/README.md:19]: Does it make sense to provide a link to a template here (or link to the corresponding tutorial section) ?

File: workflows/metabomics-lcms-preprocessing/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+**Formats:** open format as mzXML, mzMl, mzData and netCDF
+
+## Main steps
+1. MSnbase readMSData: read the mzXML and prepare for xcms
+2. XCMS findChromPeaks: peak picking
+3. XCMS groupChromPeaks: determining shared ions across samples
+4. XCMS adjustRtime: retention time correction
+5. XCMS fillChromPeaks: integrating areas of missing peaks
+6. CAMERA.annotate: annotation
+
```
Comment (mvdbeek) [inline workflows/metabomics-lcms-preprocessing/README.md:38]:
```suggestion
```

File: workflows/metabomics-lcms-preprocessing/Mass_spectrometry__LC-MS_preprocessing_with_XCMS-test.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      - class: File
+        identifier: QC1_014.mzML
+        location: https://zenodo.org/record/10130758/files/QC1_014.mzML
+        filetype: mzml
+  outputs:
+    xcms fillChromPeaks dataMatrix:
+      file: test-data/xset.merged.groupChromPeaks.adjustRtime.groupChromPeaks.fillpeaks.dataMatrix.tsv 
+    CAMERA.annotate variableMetadata:
+      file: test-data/xset.merged.groupChromPeaks.adjustRtime.groupChromPeaks.fillChromPeaks.annotate.variableMetadata.tsv
+
```
Comment (mvdbeek) [inline workflows/metabomics-lcms-preprocessing/Mass_spectrometry__LC-MS_preprocessing_with_XCMS-test.yml:65]:
```suggestion
```

**Review by mvdbeek — COMMENTED** [review]
Would it make sense to create a metabolomics folder under workflows/ ?
So your first workflow would then be in the `workflows/metabolomics/lcms-preprocessing` ?

File: workflows/metabomics-lcms-preprocessing/README.md (hunk 3 above)
Comment (lecorguille) [reply workflows/metabomics-lcms-preprocessing/README.md:19]: As explained the line after, the template can be generated with the xcms get a sampleMetadata file tool.

The tool will fill the samplename based on the infile names.

Comment (lecorguille) [reply workflows/metabomics-lcms-preprocessing/README.md]: > Would it make sense to create a metabolomics folder under workflows/ ?
Done 👍

**Comment by lecorguille** [comment]
How can I generate a [.workflowhub.yml](https://github.com/galaxyproject/iwc/blob/main/workflows/proteomics/openms-metaprosip/.workflowhub.yml)?

**Comment by mvdbeek** [comment]
You don't have to, that's autogenerated

**Comment by lldelisle** [comment]
Could you rename the '-test.yml' to '-tests.yml' (the documentation is wrong, I just realized) and regenerate the `.dockstore.yml`, please? (`planemo dockstore_init .`)
For the moment, there is no test in the dockstore so the CI does not test.

**Comment by lldelisle** [comment]
You need to regenerate it after having renamed your test, this way, it will add something like:
https://github.com/galaxyproject/iwc/blob/7fbbd2a3485c79b56c48f3745f44dfe8cd60042c/workflows/epigenetics/atacseq/.dockstore.yml#L7-L8

**Comment by lldelisle** [comment]
Apparently the renaming was sufficient to run the CI but it is better is you tell dockstore that you have a test file.

**Comment by lecorguille** [comment]
Oups, sorry, I missed the re in regenerate

File: workflows/metabomics/lcms-preprocessing/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+A sample metadata file contains various information for each of your raw files:
+- Classes which will be used during the preprocessing steps
+- Analytical batches which will be useful for a batch correction step, along with sample types (pool/sample) and injection order
+- Different experimental conditions which can be used for statistics
+- Any information about samples that you want to keep, in a column format
+
+The content of your sample metadata file has to be filled by you, since it is not contained in your raw data. Note that you can either:
+- Upload an existing metadata file
+- Use a template to create one (because it can be painful to get the sample list without misspelling or omission)
+  - Generate a template with the xcms get a sampleMetadata file tool tool
```
Comment (lldelisle) [inline workflows/metabomics/lcms-preprocessing/README.md:20]:
```suggestion
  - Generate a template with the `xcms get a sampleMetadata file` tool
```

**Comment by lldelisle** [comment]
After this small correction in the README, it is good for me also.

**Comment by lldelisle** [comment]
Youhou! I am super happy to have new workflows in IWC.

**Pull request merged**